src/game.h

```diff
@@ -443,8 +443,8 @@ class game
 
         void interactive_inv();
 
-        int inv_for_filter( const std::string &title, item_filter filter, const std::string &none_message = "" );
-        int inv_for_filter( const std::string &title, item_location_filter filter, const std::string &none_message = "" );
+        int inv_for_filter( const std::string &title, const item_filter &filter, const std::string &none_message = "" );
```

Comment:
Noting that the consumer is passed (does not have to construct) the argument it's an absolute triviality to go from `item_location` to `item`
```
loc->is_ammo()`
it.is_ammo()
```

Providing both doubles the size of your API. Size is already a problem for this PR and this is a very obvious option for reducing it.